discussions-server/src/main/java/org/xwiki/contrib/discussions/script/DiscussionsScriptService.java

```diff
@@ -226,6 +231,26 @@ public Message createMessage(String content, String syntax, String reference,
         }
     }
 
+    /**
+     * Retrieve the reference of the next message to be created: this reference should be used for temporary
+     * attachments uploads.
+     *
+     * @param serializedDiscussionReference the discussion for which to obtain a new message holder reference
+     * @param storeConfigurationParameters the related configuration parameters
+     * @return a reference to be used for displaying the message editor
+     * @since 2.1
+     * @see MessageHolderReferenceService
+     */
+    @Unstable
+    public DocumentReference getNextMessageHolderReference(String serializedDiscussionReference,
```

Comment:
Can't we make `serializedDiscussionReference` a `DocumentReference` and let Velocity do the resolution?